test/framework/src/test/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertionsTests.java

```diff
@@ -52,4 +59,172 @@ public void writeTo(StreamOutput out) throws IOException {
+                }
+                otherBuilder.endObject();
+            }
+            otherBuilder.endObject();
+            AssertionError error = expectThrows(AssertionError.class,
+                    () -> assertToXContentEquivalent(builder.bytes(), otherBuilder.bytes(), builder.contentType()));
+            assertThat(error.getMessage(), containsString("f2: expected [value2] but was [differentValue2]"));
+        }
+        {
+            XContentBuilder builder = JsonXContent.contentBuilder();
+            builder.startObject();
+            {
+                builder.startArray("foo");
+                {
+                    builder.value("one");
+                    builder.value("two");
+                    builder.value("three");
+                }
+                builder.endArray();
+            }
+            builder.field("f1", "value");
+            builder.endObject();
+
+            XContentBuilder otherBuilder = JsonXContent.contentBuilder();
+            otherBuilder.startObject();
+            {
+                otherBuilder.startArray("foo");
+                {
+                    otherBuilder.value("one");
+                    otherBuilder.value("two");
+                    otherBuilder.value("four");
+                }
+                otherBuilder.endArray();
+            }
+            otherBuilder.field("f1", "value");
+            otherBuilder.endObject();
+            AssertionError error = expectThrows(AssertionError.class,
+                    () -> assertToXContentEquivalent(builder.bytes(), otherBuilder.bytes(), builder.contentType()));
+            assertThat(error.getMessage(), containsString("2: expected [three] but was [four]"));
+        }
+        {
+            XContentBuilder builder = JsonXContent.contentBuilder();
+            builder.startObject();
+            {
+                builder.startArray("foo");
+                {
+                    builder.value("one");
+                    builder.value("two");
+                    builder.value("three");
+                }
+                builder.endArray();
+            }
+            builder.endObject();
+
+            XContentBuilder otherBuilder = JsonXContent.contentBuilder();
+            otherBuilder.startObject();
+            {
+                otherBuilder.startArray("foo");
+                {
+                    otherBuilder.value("one");
+                    otherBuilder.value("two");
+                }
+                otherBuilder.endArray();
+            }
+            otherBuilder.endObject();
+            AssertionError error = expectThrows(AssertionError.class,
+                    () -> assertToXContentEquivalent(builder.bytes(), otherBuilder.bytes(), builder.contentType()));
+            assertThat(error.getMessage(), containsString("expected [1] more entries"));
+        }
+    }
+
+    private static void addRandomNumberOfFields(XContentBuilder builder) throws IOException {
```

Comment:
No, I just forgot we had that. I will use RandomObjects.randomStoredFieldValues and only use one part of the tuple.